docs/ceph-csi-upgrade.md

```diff
@@ -0,0 +1,334 @@
+the update strategy of the CSI nodeplugin daemonsets to OnDelete so that you
+can control when the CSI driver pods are restarted on each node.
+
+To avoid this issue in future upgrades, we recommend that you do not use the
+fuse client or rbd-nbd as of now.
+
+#### 3. Upgrade CephFS Nodeplugin resources
+
+upgrading nodeplugin resources include updating the nodeplugin RBAC and
+nodeplugin daemonset
+
+##### 3.1 Update the CephFS Nodeplugin RBAC
+
+```bash
+$]kubectl apply -f deploy/cephfs/kubernetes/v1.14+/csi-nodeplugin-rbac.yaml
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+serviceaccount/cephfs-csi-nodeplugin configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+clusterrole.rbac.authorization.k8s.io/cephfs-csi-nodeplugin configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+clusterrole.rbac.authorization.k8s.io/cephfs-csi-nodeplugin-rules configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+clusterrolebinding.rbac.authorization.k8s.io/cephfs-csi-nodeplugin configured
+```
+
+**Note:** Ignore the warning messages
+
+If you determined in step 2 that you were affected by the CSI driver restart
+issue that disconnects the application pods from their mounts, continue with
+this section. Otherwise, you can skip to step 3.2
+
+```console
+vi deploy/cephfs/kubernetes/v1.14+/csi-cephfsplugin.yaml
+```
+
+```yaml
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: csi-cephfsplugin
+spec:
+  selector:
+    matchLabels:
+      app: csi-cephfsplugin
+  updateStrategy:
+    type: OnDelete
+  template:
+    metadata:
+      labels:
+        app: csi-cephfsplugin
+    spec:
+      serviceAccount: cephfs-csi-nodeplugin
+```
+
+##### 3.2 Update the CephFS Nodeplugin daemonset
+
+```bash
+[$]kubectl apply -f deploy/cephfs/kubernetes/v1.14+/csi-cephfsplugin.yaml
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+daemonset.apps/csi-cephfsplugin configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+service/csi-metrics-cephfsplugin configured
+```
+
+**Note:** Ignore the warning messages
+
+##### 3.3 Manual deletion of CephFS Nodeplugin daemonset pods
+
+If you determined in step 2 that you were affected by the CSI driver restart
+issue that disconnects the application pods from their mounts, continue with
+this section. Otherwise, you can skip this section.
+
+As we have set the updateStrategy to OnDelete the CSI driver pods will not be
+updated until you delete them manually. This allows you to control when your
+application pods will be affected by the CSI driver restart.
+
+For each node:
+
+- Drain your application pods from the node
```

Comment:
Should we add commands and examples for actions presented here? (if agreed to, applies to the rbd section as well)

Reply:
we can leave this one, it's up to the user how they want to drain nodes in their cluster.